crates/jstz_core/src/native.rs

```diff
@@ -455,13 +447,25 @@ pub trait NativeClass {
     /// The attributes the class will be bound with, default is `writable`, `enumerable`, `configurable`.
     const ATTRIBUTES: Attribute = Attribute::all();
 
-    /// The constructor of the class.
-    fn constructor(
-        this: &JsNativeObject<Self::Instance>,
+    /// Creates the internal data for an instance of this class.
+    fn data_constructor(
+        target: &JsValue,
```

Comment:
target is never used in the class that implements it, i'm curous how we would ever need it?

Reply:
If you used static properties of the target in the initializer